src/app/shared/components/km-select/template.html

```diff
@@ -0,0 +1,28 @@
+<!--
+Copyright 2022 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<form [formGroup]="form">
+  <mat-form-field>
+    <mat-label>{{options.length > 0 ? label : 'no '+label+' Available'}}</mat-label>
```

Comment:
better to use *ngIf as done on other shared components
```suggestion
<mat-label *ngIf="label">{{label}}</mat-label>
```